### PR TITLE
Bluetooth: controller: Workaround multi-link DLE assert

### LINKED ISSUE #11841: Assert invoked in exception in ctrl.c line 7653 (zephyr 1.13.00)
Assert gets invoked when having one connection and starting another. Spoke to @cvinayak about this and it seems the new connection wants to resize the rx_pool while it's still in use by the first connection. 
`if (_radio.packet_rx_data_count != free_count_rx) {
  /** TODO another role instance has obtained
    * memory from rx pool.
    */
    //LL_ASSERT(0);
} `

### PR DESCRIPTION
**Pull request by cvinayak**
Workaround from asserting when Rx PDU buffers from pool
needing resize is acquired by another connection. By
skipping the connection events to give some headroom for
the host to process the Rx packets and eventually make the
pool available for resize.

Fixes #11841.

Signed-off-by: Vinayak Kariappa Chettimada <vich@nordicsemi.no>

### CODE REVIEW

**Comment by cvinayak** [comment]
@alexsven could you verify if this is a suitable workaround in comparison to commenting out the assert?

File: subsys/bluetooth/controller/ll_sw/ctrl.c [hunk 1]
```diff
@@ -8007,10 +8007,12 @@ static inline void event_len_prep(struct connection *conn)
...
 		if (_radio.packet_rx_data_count != free_count_rx) {
-			/** TODO another role instance has obtained
-			 * memory from rx pool.
+			/* NOTE: Another role instance has obtained
+			 * memory from rx pool, skip this event and
+			 * check in the next event if resize can be
+			 * done.
 			 */
-			LL_ASSERT(0);
+			return -EAGAIN;
```
Comment (jhedberg) [inline subsys/bluetooth/controller/ll_sw/ctrl.c]: This doesn't look right. You changed the return value of this function to u32_t but here you're returning a negative value?

**Review by jhedberg — CHANGES_REQUESTED** [review]

File: subsys/bluetooth/controller/ll_sw/ctrl.c (hunk 1 above)
Comment (cvinayak) [reply subsys/bluetooth/controller/ll_sw/ctrl.c]: Oversight, will fix it. Thank you.

**Review by jhedberg — APPROVED** [review]

**Review by thoh-ot — APPROVED** [review]
QUICK-ACK

**Comment by alexsven** [comment]
> @alexsven could you verify if this is a suitable workaround in comparison to commenting out the assert?
@cvinayak 
Verified that it is working here!

**Review by carlescufi — APPROVED** [review]

**Pull request merged**